packages/cyberstorm/src/components/DropDown.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { ReactElement, ReactNode, useState } from "react";
+import styles from "./componentStyles/DropDown.module.css";
+import { Button } from "./Button";
+import { faChevronDown } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+import * as DropdownMenu from "@radix-ui/react-dropdown-menu";
+import { DropDownItemProps } from "./DropDownItem";
+
+type DropDownProps = {
+  icon?: ReactNode;
+  defaultOpen?: boolean;
+  options?: DropDownOptions;
+  colorScheme?: "default" | "defaultDark" | "primary";
+  defaultSelectedOptionKey?: number;
+};
+
+export type DropDownOption = {
+  label: string;
+  content?: ReactElement<DropDownItemProps>;
+};
+export type DropDownOptions = Map<number | string, DropDownOption>;
+
+export const DropDown: React.FC<DropDownProps> = (props) => {
+  const { icon, colorScheme, defaultOpen, options, defaultSelectedOptionKey } =
+    props;
+
+  let initialValue: DropDownOption | null =
+    options?.values().next().value ?? null;
+
+  if (
+    options &&
+    defaultSelectedOptionKey &&
+    options.get(defaultSelectedOptionKey)
+  ) {
+    initialValue = options.get(defaultSelectedOptionKey) ?? null;
+  }
+
+  const [value, setValue] = useState(initialValue);
```

Comment:
Don't know how I missed this in the previous review, but do we actually do anything with the `value`? I.e. what happens the value changes? AFAIK currently the `DropDown`'s internal state is updated, but then what? Does the parent page/component have anyway to access the value or detect the change? Does Radix handle this somehow behind the scenes?

Reply:
Not yet used, but when the component is used on a form etc. the setValue could instead be a callback function..... anyways I don't quite know what's the best way for managing the state.
The DropDown component has to update (label on button changes) based on the selected value and the parent component should also be aware of the selected value (state).

Reply:
I mean I don't know if it is good to implement the state management on the storybook side, but also that would be the logical place to manage it.

Reply:
Yes I think it's a good idea to implement the management outside the component (e.g. in Storybook) unless we have a reason to do it otherwise.